ide/coqide/idetop.ml

```diff
@@ -360,6 +360,59 @@ let debug_cmd = ref ""
 let db_cmd cmd =
   debug_cmd := cmd
 
+let db_loc () =
+  let open Loc in
+  let open DebugHook in
+  match debugger_state.cur_loc with
+  | Some {fname=ToplevelInput; bp; ep} ->
+    Some ("ToplevelInput", [bp; ep])
+  | Some {fname=InFile {dirpath=None; file}; bp; ep} ->
+    Some (file, [bp; ep])  (* for Load command *)
+  | Some {fname=InFile {dirpath=Some dirpath}; bp; ep} ->
+    (* todo: check what's in Loc.t on Windows *)
+    let open Names in
+    let dirpath = DirPath.of_string dirpath in
```

Comment:
What's the point of creating `dirpath` with a hackish parsing function that shouldn't even be in the kernel to start with, when the only thing you do with it below is to perform string manipulation? Please remove the kernel function and just use simple functions for lists of identifiers.